src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -414,6 +414,29 @@ JNI_METHOD(jstring, getIpAddress)(JNIEnv * env, jobject self, jlong handle, jlon
     return env->NewStringUTF(addrStr);
 }
 
+JNI_METHOD(void, updateAddress)(JNIEnv * env, jobject self, jlong handle, jlong deviceId, jstring address, jint port)
+{
+    ScopedPthreadLock lock(&sStackLock);
+    Device * chipDevice = nullptr;
+    CHIP_ERROR err      = CHIP_NO_ERROR;
+
+    GetCHIPDevice(env, handle, deviceId, &chipDevice);
+
+    Inet::IPAddress ipAddress = {};
+    JniUtfString addressAccessor(env, address);
+    VerifyOrExit(Inet::IPAddress::FromString(addressAccessor.c_str(), ipAddress), err = CHIP_ERROR_INVALID_ADDRESS);
+    VerifyOrExit(CanCastTo<uint16_t>(port), err = CHIP_ERROR_INVALID_ADDRESS);
+
+    err = chipDevice->UpdateAddress(Transport::PeerAddress::UDP(ipAddress, port));
```

Comment:
This only accounts for IPv4, since it doesn't check the address scope, so that if a Link-local address appears in a record, (e.g. for an IPv6 wifi device), that it will select the right zone index to target the interface.

Reply:
I don't understand why it only accounts for IPv4. The code uses whatever address a given node publishes. In case it's a link-local address isn't it a job of a message sender to use the right interface to reach that address?

Reply:
Also, it's a problem of other platforms as well and it hasn't yet been determined how it's to be solved: https://github.com/project-chip/connectedhomeip/issues/7232.

Reply:
Note that #7232 was solved by actually using the interface returned from mdns query machinery.  Basically, for an ipv6 link-local address you have to remember which interface your mdns response came in on and make sure you use that one to talk to it.

In terms of the APIs here, that means that the interface to use needs to be one of the args to `updateAddress`; there's no way to recover that information from inside the implementation of `updateAddress`.  @Damian-Nordic